tests/06-enums.rs

```diff
@@ -39,12 +36,18 @@
 //         Thirteen = 0b1101,
 //     }
 //
-//     ...
+// then the number of bits required to represent the struct is coerced to N.
+// Calling a getter on the struct returns a Result<T, EnumError>, where an
+// EnumError has an UnknownValue variant that can provide the raw value.
+//
 //     let mut bitfield = MyBitfield::new();
 //     assert_eq!(0, bitfield.small_prime().unwrap_err().raw_value());
 //
 //     bitfield.set_small_prime(SmallPrime::Seven);
 //     let p = bitfield.small_prime().unwrap_or(SmallPrime::Two);
+//
+// The getter for an enum marked with the bits attribute will return a Result
+// even if the struct has a number of variants which is a power of two.
```

Comment:
Why was this decision made? Can it be useful that way? Couldn't we do better since the proc. macro should be able to find out if `N` is a power-of-two since we can query this information easily at compile time. The only downside I see to this is extensibility so that you cannot easily extend the enum to a non-power-of-two variants later without breaking dependencies. However, there is an easy trick against this by simply putting another guardian enum in place. Also it will probably break dependencies anyways if its bitfield size is changed.

Reply:
The situation I'm thinking of is one where somebody has written an enum like this:
```
#[derive(BitfieldSpecifier)]
#[bits = 2]
enum PwrMgmtMode {
   OnOff = 0b00,
   CyclicTracking = 0b01,
   Sleeping = 0b10,
}
```
(with `0b11` reserved)

Then somebody decides they need to add the last one:
```
    Sleeping = 0b10,
+   SomeFourthMode = 0b11,
 }
```
they should be able to make that change without having to change everywhere they use a getter to no longer `.unwrap()`.

(of course, one could argue that they _should_ take the opportunity to stop checking the now never-throwing errors, but there's no harm per se if they do check, and it may be a lot of code to refactor. I suspect the compiler should optimize it out anyway)

Reply:
I see your point. This is indeed bad for usability of the crate.
For me the purpose of `[bits = N]` is two fold:
    1. To tell the compiler how many bits we actually want to have. Normally this is going to be the minimum amount of bits necessary to store all the information required. E.g. 2 bits for 1-4 variants etc.
    2. Also I would probably use it as additional safety guard that we won't go over that limit by accident. The amount of bits it entails is also a breaking factor for dependencies. One could argue that the `#[bitfield]` macro already has this safety guard in place for it but why not both?

I personally would always try to generate the best possible code with least possible friction. For me this means that we would only return `Result` if we really need to. However, I definitely see the point you have made and it is concerning.

Another idea might be to introduce yet another attribute with which a user can control the behavior of this. E.g. `#[specifier(return_result)]` (I am bad with names).

Reply:
Hmm... I think I like the specifier idea (some names... `#[may_fail]`? `#[not_complete]`? `#[returns_error]`?). If the specifier is present, then behaviour like in this PR as-is, if the specifier is not present, then compilation failure if there are not `2**bits` variants. `#[may_fail]` without a `#[bits = N]` is fine, it just changes the getter to return a result and auto-detects the bits required.

It's a bit more verbose, but it separates the "non-power-of-two" behaviour from the "what do getters return" behaviour.

And if the specifier were named something like `#[not_complete]` or `#[round_up_bits]`, then non-power-of-two enums could be specified and the number of bits could be inferred. (I'm not sure I see an actual use case for this though)